tests/main/snap-user-service-start-on-install/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Check that snap user services are installed and started.
```

Comment:
this test and the snap-user-service seems to have the same summary? what's the difference?

do we need a test that the right thing happens at refresh? stopping and starting for old and new version of a user services?  probably also one where the new version fails to start?

Reply:
I've added an extra spread test that demonstrates that running user daemons are restarted when upgrading a snap.

Reply:
I've added a second spread test that tries to install an upgrade with a failing user daemon, showing that the original revision's daemon has been restarted after rolling back the upgrade.

Interestingly, if there is no user session running during the refresh then the upgrade will succeed.  I'm not sure there is anything that can be done about that.